src/Compilers/CSharp/Portable/Syntax/SyntaxEquivalence.cs

```diff
@@ -43,19 +44,15 @@ public static bool AreEquivalent(SyntaxTokenList before, SyntaxTokenList after)
 
         public static bool AreEquivalent(SyntaxToken before, SyntaxToken after)
         {
-            return before.RawKind == after.RawKind && (before.Node == null || AreTokensEquivalent(before.Node, after.Node));
+            return before.RawKind == after.RawKind && (before.Node == null || AreTokensEquivalent(before.Node, after.Node, ignoreChildNode: null));
         }
 
-        private static bool AreTokensEquivalent(GreenNode before, GreenNode after)
+        private static bool AreTokensEquivalent(GreenNode before, GreenNode after, Func<SyntaxKind, bool> ignoreChildNode)
```

Comment:
can you help me see who passes in a non-null value for this func?

Reply:
It's in the public API: [http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/SyntaxFactory.cs,95b5ee641c05da91,references](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/SyntaxFactory.cs,95b5ee641c05da91,references)

---
In reply to: [333780400](https://github.com/dotnet/roslyn/pull/39227#discussion_r333780400) [](ancestors = 333780400)

Reply:
@CyrusNajmabadi for even more reference, the IDE now needs to use this func to ignore directives in the import service :).